src/dom-to-image-more.js

```diff
@@ -1199,13 +1199,31 @@
                 sandbox.style.visibility = 'hidden';
                 sandbox.style.position = 'fixed';
                 document.body.appendChild(sandbox);
-                // Ensure that the iframe is rendered in standard mode
+                sandbox.id = sandbox.contentDocument.title = 'domtoimage-sandbox';
+                // Ensure the iframe's character rendering matches the document (with UTF-8 fallback).
                 const charset = document.createElement('meta');
                 charset.setAttribute('charset', document.characterSet || 'UTF-8');
                 sandbox.contentDocument.head.appendChild(charset);
-                sandbox.contentDocument.title = 'sandbox';
+                // Ensure the document is rendered in standard mode by coercing doctype.
+                // If the parent document has a Trusted Types policy, fallback to quirks mode.
+                let isTrustedTypesRequired;
+                try {
+                    const xhr = new XMLHttpRequest();
+                    xhr.open('GET', document.location.href, false);
+                    xhr.send();
+                    const csp = xhr.getResponseHeader('content-security-policy');
+                    isTrustedTypesRequired = csp.indexOf('require-trusted-types-for') !== -1;
+                } catch(_) {
+                    isTrustedTypesRequired = true;
+                }
+                if (!isTrustedTypesRequired) {
+                    const sandboxDocument = document.implementation.createHTMLDocument(sandbox.id);
+                    sandboxDocument.head.appendChild(charset);
+                    const sandboxDoctype = document.doctype ? '<!DOCTYPE html>' : '';
+                    const sandboxHTML = `${sandboxDoctype}${sandboxDocument.documentElement.outerHTML}`;
+                    sandbox.setAttribute('srcdoc', sandboxHTML);
+                }
```

Comment:
So, the question I have is what does this actually do versus the original code? Seems like the path where we don't need trusted types is handled now, but what happens if we _do need them_?

Reply:
If the page would have thrown an error because of Trusted Types, it'll have the `iframe` with the #100 regression present. If it doesn't have a Trusted Types policy - most pages thankfully don't, we do go ahead and coerce in the doctype.

Reply:
Seems icky to be doing the DOCTYPE handling only in one case, but I guess because we're creating the IFRAME from the main document, it gets the right one inside?

Reply:
Yep. HTML is the correct doctype for our default style query seeing as we embed inside a HTML `foreignObject`

Reply:
My thought (so poorly expressed, sorry), is that the net effect of this code when the `isTrustedTypesRequired === true` is that we're not actually setting the DOCTYPE of the new `IFRAME` at all. I am wondering if we can/should... seems that the DOM `.doctype` is read-only, so I doubt we can change anything since we're already synthesized the` iframe` up at line 1198. I guess this PR (and the one that led to it) has gone through enough churn that I am a bit gunshy about this whole code segment. Sorry.

I might be a bit less worried if we were doing _anything_ in the (non-existent) `else` clause for this test... in that my mind could settle on what's supposed to be happening in either case, but I'm too close.